# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/WebCodecsUtils.h
# Commit: 0899b9a2539e
# Full Hash: 0899b9a2539e6d7f98385c2997a3b7d8d5ec5596
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 2
# Description:
#   Bug 1831451 - Guess color space from VideoData r=aosmond
#   
#   Depends on D171940
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D170635
# ==============================================================================

diff -r 290fcbe92f0a -r 0899b9a2539e dom/media/webcodecs/WebCodecsUtils.h
--- a/dom/media/webcodecs/WebCodecsUtils.h	Thu Jun 29 22:33:13 2023 +0000
+++ b/dom/media/webcodecs/WebCodecsUtils.h	Thu Jun 29 22:33:13 2023 +0000
@@ -11,6 +11,7 @@
 
 #include "ErrorList.h"
 #include "js/TypeDecls.h"
+#include "mozilla/Maybe.h"
 #include "mozilla/Result.h"
 #include "mozilla/Span.h"
 #include "mozilla/dom/UnionTypes.h"
@@ -68,6 +69,16 @@
 
 gfx::ColorSpace2 ToPrimaries(VideoColorPrimaries aPrimaries);
 
+bool ToFullRange(const gfx::ColorRange& aColorRange);
+
+Maybe<VideoMatrixCoefficients> ToMatrixCoefficients(
+    const gfx::YUVColorSpace& aColorSpace);
+
+Maybe<VideoTransferCharacteristics> ToTransferCharacteristics(
+    const gfx::TransferFunction& aTransferFunction);
+
+Maybe<VideoColorPrimaries> ToPrimaries(const gfx::ColorSpace2& aColorSpace);
+
 }  // namespace dom
 }  // namespace mozilla
 